File: 283-Move-Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int left_pointer = 0;
        for(int right_pointer = 0;right_pointer<nums.length;right_pointer++){
            if(nums[right_pointer]!=0){
                int temp;
                temp = nums[right_pointer];
                nums[right_pointer] = nums[left_pointer];
                nums[left_pointer] = temp;
                left_pointer++;
            }
        }
    }
}

File: 283-Move-Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int leftPointer=0;
        int rightPointer=0;
        for(int rightPointer=0; rightPointer<nums.size(); rightPointer++ ){
            if(nums[rightPointer] != 0){
                swap(nums[leftPointer],nums[rightPointer]);
                leftPointer++;
            }
        }
    }
};